### ISSUE TITLE
Look up by second level domain.

### ISSUE THREAD

**Issue by developer91234** [keepassxreboot/keepassxc-browser#1309]
This is what all other password fillers do since sites change their login entry-points often. If I have a mail.google.com account, other fillers will fill it on all google sites. Keepass browser will only fill it on mail.google.com (which is no longer used by google or logins). I don't see the use case for looking up by complete domain name.

**Comment by varjolintu** [comment]
An example:
Entry URL `https://google.com` will fill all subdomains under `google.com`.
Entry URL `https://mail.google.com` will fill that subdomain plus all other subdomains under `mail.google.com`.

What are you meaning here exactly?

**Comment by developer91234** [comment]
The `mail.google.com` entry should fill everything under `google.com`

The problem is that other password managers save the subdomain (as does keepassxc by default). But many websites have different login entry points, which moreover change. To get around this issue you have to ignore subdomains in fetching entries, which is the default behavior of most fillers.

**Comment by varjolintu** [comment]
That's not how the matching works. If you wish to match everything under `google.com` you must use `https://google.com` as your entry URL. It works as a wildcard similar to `https://*.google.com`.

**Comment by developer91234** [comment]
That's not how it works on Keepass, but it is how it works everywhere else and for a very good reason. I have a bunch of credentials that I saved years ago, and the sites have since changed the entry points. What is the user supposed to do this in this case, write a script to strip away the subdomains?

**Comment by droidmonkey** [comment]
We are not going to change this behavior because it is in place for very good reasons.

**Comment by droidmonkey** [comment]
We support multiple urls per entry, so if you want you can define google.com as an additional url and still have the specific site link as the main url. We aren't changing the behavior so you can modify the code yourself if you want.

**Comment by developer91234** [comment]
I was hoping to hear a plausible use-case for the current behavior, but OK I'll take no for an answer.

**Comment by droidmonkey** [comment]
The obvious use case is anyone who wants credentials defined for specific urls. By going to the higher level domain you nullify that option entirely.

**Comment by developer91234** [comment]
Yeah that's the use case for a person who logs into different portals on a given top-level domain, but that use case is extremely rare (and is also still more or less satisfied by using top-level domains, by simply using the selection dialog for that very rare occasion). On the other hand, the far more common use-case of someone with a dozens of obsolete or overly-specific urls in their database is not satisfied at all (except through scripting or manual editing).

Password-managers save the login urls only. Login urls change all the time, only the top-level domain is stable. So pretty much every long-time user will eventually get some obsolete/unusable entries in their database and won't be able to login using the URL.

Anyway, that's was the rationale for at least creating an option to look up by top-level domain. I think it's overwhelming. But it's your decision so I won't press the matter further.

**Comment by droidmonkey** [comment]
I am certainly one of those users that have obsolete urls, I find them all the time. I update them when I find them though. I could see an application setting that is basically the opposite of return best matching credentials. 

I'll reopen this with that thought.
